test/third-party/MatchPattern.test.js

```diff
@@ -1,10 +1,211 @@
 describe('third-party/MatchPattern', () => {
-  const matches
-      = (pattern, urlStr) => assert.isOk(pattern.doMatch(new URL(urlStr)));
-  const notMatches
-      = (pattern, urlStr) => assert.isNotOk(pattern.doMatch(new URL(urlStr)));
+    describe('invalid patterns', () => {
+      const newPattern =
+        pattern => { return () => new MatchPattern(pattern) };
```

Comment:
I think this fits on one line, but for all statements continued across >1 line, continuations should be indented 4 spaces.